### PR TITLE
CDAP-9084 If run id does not exist while querying run level logs then return 404 instead of 500.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-9084

### CODE REVIEW

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java [hunk 1]
```diff
@@ -245,16 +246,34 @@ public void testWorkflowLogsPrev() throws Exception {
                   ImmutableList.of("logLevel", "threadName"));
   }
 
+
   private List<LogLine> getLogs(String namespaceId, String appId, String programType, String programName, String runId,
-                                String endPoint) throws Exception {
+                                String endPoint, int statusCode) throws Exception {
```
Comment (anwar6953) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java]: rename to `expectedStatusCode`?

Comment (sagarkapare) [reply cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java]: fixed.

**Comment by anwar6953** [comment]
LGTM

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-RUT821-1

**Comment by sagarkapare** [comment]
Build is green - https://builds.cask.co/browse/CDAP-RUT821-2. Merging.

**Pull request merged**